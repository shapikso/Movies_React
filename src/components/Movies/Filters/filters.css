@keyframes active {
    from { transform: scale(0); }
    to { transform: scale(1); }
}

.form-filters {
    position: absolute;
    top: 100px;
    padding: 40px;
    -webkit-backdrop-filter: blur(50px);
    backdrop-filter: blur(50px);
    min-width: 560px;
    max-width: 1020px;
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;

    max-height: 90%;
    overflow-y: auto;
}

.form-filters::-webkit-scrollbar {
    display: none;
}

.form-filters__header {
    display: flex;
    justify-content: space-between;
    flex-wrap: wrap;
}

.form-filters__header >:not(:last-child) {
    margin-right: 25px;
}

.form-filters__header * {
    margin-bottom: 40px;
}

.form-filters select {
    -webkit-appearance: none;
    -moz-appearance: none;
    appearance: none;
    /*background: url("../../../filters/arrow-down.svg") no-repeat 15px center;*/
    padding-left: 40px;
}

.form-filters select:hover {
    cursor: pointer;
}

.form-filters select option {
    color: var(--dark-primary-color);
}

.form-filters__ranges {
    display: flex;
    justify-content: space-between;
    flex-wrap: wrap;
}

.form-filters__ranges >:not(:last-child) {
    margin-right: 20px;
}

.form-filters__ranges__item {
    margin-bottom: 40px;
}

.double-range {
    border: 1px solid var(--light-primary-color);
    border-radius: 5px;
    padding: 14px;
}

.double-range >:not(:last-child) {
    margin-bottom: 15px;
}

.double-range__info {
    width: 250px;
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.double-range__info-min,
.double-range__info-max {
    width: 97%;
    background: none;
    color: var(--light-primary-color);
    padding: 10px;
    text-align: center;
}

.double-range__info-max {
    text-align: right;
}

.double-range__place {
    width: 100%;
    position: relative;
    height: 20px;
}

.double-range__place:before {
    content: "";
    width: 100%;
    height: 2px;
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
    left: 0;
    background-color: var(--light-primary-color);
}

.double-range__place-item {
    position: absolute;
    left: 0;
    top: 50%;
    transform: translateY(-50%);
    -webkit-appearance: none;
    -moz-appearance: none;
    appearance: none;
    width: 100%;
    height: 0;
    padding: 0;
    margin: 0;
    border: none;
    outline: none;
    background: none;
}

.double-range__place-item::-webkit-slider-runnable-track {
    width: 100%;
    height: 0;
    cursor: pointer;
    background-color: var(--light-primary-color);
    box-shadow: none;
    border: none;
    outline: none;
    z-index: 0;
}

.double-range__place-item::-webkit-slider-thumb {
    z-index: 2;
    position: relative;
    height: 18px;
    width: 18px;
    border-radius: 50%;
    background-color: var(--light-primary-color);
    cursor: pointer;
    -webkit-appearance: none;
    appearance: none;
    margin-top: -8px;
}

.double-range__place-item::-moz-range-track {
    width: 100%;
    height: 0;
    cursor: pointer;
    background-color: var(--light-primary-color);
    box-shadow: none;
    border: none;
    outline: 0;
    margin: 0;
    padding: 0;
    z-index: 0;
}

.double-range__place-item::-moz-range-thumb {
    z-index: 2;
    position: relative;
    height: 18px;
    width: 18px;
    border-radius: 50%;
    border: none;
    background-color: var(--light-primary-color);
    cursor: pointer;
}

.form-filters__release-date {
    display: flex;
    justify-content: space-between;
    margin-bottom: 40px;
    width: 880px;
}

.form-filters__release-date__item {
    width: 48%;
}

.form-filters__buttons {
    display: flex;
    justify-content: center;
    position: relative;
    gap: 40px;
}

.form-filters__buttons button[type="reset"] {
    min-width: 0;
    max-width: 100px;
    width: 100%;
    position: absolute;
    top: 0;
    right: -285px;
}

.date-picker {
    position: relative;
}

.date-picker__input {
    width: 100%;
}

.date-picker__input[type="text"]:focus {
    box-shadow: none;
}

.date-picker__clear {
    width: 50px;
    height: 50px;
    position: absolute;
    top: 0;
    right: 0;

    display: flex;
    justify-content: center;
    align-items: center;
    border-left: 1px solid var(--light-primary-color);
    border-radius: 0 5px 5px 0;
    transition: all 0.2s ease;
}

.date-picker__clear i {
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
}

.date-picker__clear i svg {
    display: block;
    transition: all 0.2s ease;
}

.date-picker__clear:hover i svg {
    fill: var(--primary-color);
}

.date-picker__clear:active i svg {
    fill: var(--light-primary-color);
}

.date-picker__clear:hover {
    cursor: pointer;
    background-color: var(--light-primary-color);
}

.date-picker__clear:active {
    background-color: var(--primary-color-opacity);
}

.frosted-glass-effect {
    background: linear-gradient(
            155deg
            , rgba(255, 255, 255, 0.2) -50%, rgba(255, 255, 255, 0.05) 80%);
    border: 1px solid rgba(255, 255, 255, 0.3);
    border-radius: 25px;
    -webkit-backdrop-filter: blur(10px);
    backdrop-filter: blur(10px);
}

.basic-field {
    padding: 10px 25px;
    background: none;
    border: 1px solid var(--light-primary-color);
    border-radius: 5px;
    height: 50px;
    font-size: 16px;
    color: var(--light-primary-color);
}
.sliderArea {
    width: 300px;
    margin: 0 auto;
}
.close {
    cursor: pointer;
    font-size: 30px;
    min-width: 20px;
    position: absolute;
    right: -340px;
    top: -480px;
}

