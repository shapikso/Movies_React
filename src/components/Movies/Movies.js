import React, { Component } from 'react';
import Button from '../common/Button/Button';

class Movies extends Component {
    render() {
        return (
            <div>
                <h1>Movies</h1>
                <Button/>
            </div>
        );
    }
}

export default Movies;