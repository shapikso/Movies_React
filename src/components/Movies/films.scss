.movie-wrapper {
  display: flex;
  flex-direction: column;
  height: 100%;
  position: relative;
}

.movie-wrapper::after {
  content: "";
  width: 150px;
  height: 150px;
  border-radius: 50%;
  background: linear-gradient(225.82deg, #352F44 24.16%, #5C5470 87.91%);
  position: fixed;
  top: 50px;
  left: -75px;
  z-index: -1;
}
.movie-wrapper::before {
  content: "";
  width: 150px;
  height: 150px;
  border-radius: 50%;
  background: linear-gradient(225.82deg, #352F44 24.16%, #5C5470 87.91%);
  position: fixed;
  bottom: 50px;
  right: -75px;
  z-index: -1;
}

#filter-btn {
  right: 0;
  min-width: 100px;
  height: 40px;
}
.header__links {
  display: flex;
  justify-content: space-between;
}
.center {
  display: flex;
  justify-content: center;
  margin-bottom: 40px;
}
.movies {
  margin-top: 30px;
  display: grid;
  grid-template-columns: repeat(auto-fill, 340px);
  grid-gap: 20px;
  justify-content: center;
  flex-grow: 1;
}

.not-found-fill {
  display: block;
  font-size: 50px;
  color: var(--light-primary-color-opacity);
  text-align: center;
}

.place-right {
  display: flex;
  justify-content: end;
  flex-shrink: 0;
}
.film-poster {
  background: #000;
  width: 100%;
  max-height: 200px;
  height: 100%;
  overflow: hidden;
  transition: max-height 0.5s ease;
}
.small-poster{
  width: 100%;
  height: 200px;
}
.film-description {
  width: 100%;
  max-height: 50px;
  height: 100%;
  padding: 15px;
  display: flex;
  justify-content: space-between;
  text-align: center;
  align-items: flex-end;
  background: var(--primary-color-opacity);
  transition: max-height 0.5s ease;
  overflow: hidden;
}
.card-wrapper {
  width: 100%;
  height: 100%;
  border-radius: 10px;
  overflow: hidden;
}
.card {
  width: 340px;
  height: 250px;
  font-size: 16px;
  position: relative;
  margin-bottom: 80px;
  border-radius: 10px;
}
@keyframes unscale {
  from {
    transform: scale(1.1);
  }
  to {
    transform: scale(1);
  }
}
@keyframes scale {
  from {
    transform: scale(1);
  }
  to {
    transform: scale(1.1);
  }
}
.card img {
  animation: unscale 0.5s ease;
}
.card:hover {
  cursor: pointer;
}
.card:hover img {
  animation: scale 0.5s ease forwards;
}
.film-name {
  max-width: 70%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}
.film-duration {
  max-width: 30%;
}
.film-rate {
  width: 50px;
  height: 22px;
  position: absolute;
  left: -10px;
  top: 10px;

  justify-content: center;
  align-items: center;
  padding: 4px;
  border-radius: 5px;
  text-align: center;
}
.good-rate {
  background-color: #246724;
}
.bad-rate {
  background-color: #5B5B5B;
}

.filter-active {
  background: #ffffff;
  color: #151C26;
}

.not-fond-wrapper {
  display: none;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  position: absolute;
  top: 40%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.svg-not-found {
  width: 60%;
  height: 60%;
  color: var(--light-primary-color-opacity);
}

.filters-modal-box {
  background-color: #000000a1;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 500%;
  padding: 50px;
  display: flex;
  justify-content: center;
  align-items: center;
}

.filters-modal-box.active {
  animation: active 0.2s ease forwards;
  z-index: 10;
}

.filters-modal-box.hide {
  display: none;
}
.movie-title {
  flex: 1 0 auto;
  padding-left: 41px;
  width: 500px
}