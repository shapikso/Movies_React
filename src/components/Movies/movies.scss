@import '../common/styles/colors.scss';

.movie-wrapper {
    display: flex;
    flex-direction: column;
    height: 100%;
    position: relative;
}

.movie-wrapper::after {
    content: "";
    width: 150px;
    height: 150px;
    border-radius: 50%;
    background: linear-gradient(225.82deg, $dark-primary 24.16%, $primary-color 87.91%);
    position: fixed;
    top: 50px;
    left: -75px;
    z-index: -1;
}
.movie-wrapper::before {
    content: "";
    width: 150px;
    height: 150px;
    border-radius: 50%;
    background: linear-gradient(225.82deg, $dark-primary 24.16%, $primary-color 87.91%);
    position: fixed;
    bottom: 50px;
    right: -75px;
    z-index: -1;
}

#filter-btn {
    right: 0;
    min-width: 100px;
    height: 40px;
}

.center {
    display: flex;
    justify-content: center;
    margin-bottom: 40px;
}

.movies {
    margin-top: 30px;
    display: grid;
    grid-template-columns: repeat(auto-fill, 340px);
    grid-gap: 20px;
    justify-content: center;
    flex-grow: 1;
}

.place-right {
    display: flex;
    justify-content: end;
    flex-shrink: 0;
  }