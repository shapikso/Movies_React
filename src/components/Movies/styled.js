import styled from "styled-components";
import {COLORS} from '../common/styles/colors';

export const StMovieWrapper = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
  position: relative;
  max-width: 1500px;
  padding: 0 40px;
  margin: 0 auto;
  ::after {
    content: "";
    width: 150px;
    height: 150px;
    border-radius: 50%;
    background: linear-gradient(225.82deg, ${COLORS.darkPrimary} 24.16%, ${COLORS.primaryColor} 87.91%);
    position: fixed;
    top: 50px;
    left: -75px;
    z-index: -1;
  };
  ::before {
    content: "";
    width: 150px;
    height: 150px;
    border-radius: 50%;
    background: linear-gradient(225.82deg, ${COLORS.darkPrimary} 24.16%, ${COLORS.primaryColor} 87.91%);
    position: fixed;
    bottom: 50px;
    right: -75px;
    z-index: -1;
  };
`;

export const StPlaceRight = styled.div`
  display: flex;
  justify-content: end;
  flex-shrink: 0;
`;

export const StMovies = styled.div`
  margin-top: 30px;
  display: grid;
  grid-template-columns: repeat(auto-fill, 340px);
  grid-gap: 20px;
  justify-content: center;
  flex-grow: 1;
`;

export const StCenter = styled.div`
  display: flex;
  justify-content: center;
  margin-bottom: 40px;
`;

export const StFiltersModal = styled.div`
  background-color: #000000a1;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 500%;
  padding: 50px;
  justify-content: center;
  align-items: center;
  display: ${({hidden}) => hidden ? 'none' : 'flex'};
`;

export const StOption = styled.option`
  color: #151C26;
`;
