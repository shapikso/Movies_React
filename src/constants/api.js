export const URL_IMG = 'https://image.tmdb.org/t/p/original/';
export const NEW_API_URL = 'http://localhost:3001';

export const URL_MOVIE = `${NEW_API_URL}/movies`;
export const MOVIE_ON_PAGE = 8;
