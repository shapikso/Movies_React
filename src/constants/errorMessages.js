export const MESSAGE_UPPERCASE_LETTER = 'Starts with uppercase letter';
export const MESSAGE_LATIN_LETTER = 'Only latin letters';
export const MESSAGE_MIN_CHARACTERS = 'Min two characters';
export const MESSAGE_LATIN_LETTER_AND_NUMBERS = 'Only latin letters and numbers';
export const MESSAGE_RANGE_CHARACTERS = 'From 4 to 20 characters';
export const MESSAGE_START_WITH_LETTER = 'Starts with letter';
export const MESSAGE_MIN_SYMBOLS = 'Min 8 symbols';
export const MESSAGE_MIN_NUMBERS = 'Min 1 number';
export const MESSAGE_LOWER_CASE_LETTER = 'Min one lowercase letter';
export const MESSAGE_ONE_UPPERCASE_LETTER = 'Min one uppercase letter';