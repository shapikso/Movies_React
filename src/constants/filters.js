export const FILTERS_INIT = {
    title: '',
    language: '',
    status: '',
    release_date_first: '',
    release_date_last: '',
    budget_min: 0,
    budget_max: 500000000,
    popularity_min: 0,
    popularity_max: 200,
    revenue_min: 0,
    revenue_max: 30000,
};
