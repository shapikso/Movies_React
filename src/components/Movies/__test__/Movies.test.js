import React from 'react';
import { mount } from 'enzyme';
import Movies from '../Movies';
import axios from 'axios';
import mockMovies from './__movies-mock__';
import { act } from 'react-dom/test-utils';
import { BrowserRouter } from 'react-router-dom';
import { FILTERS_INIT } from '../../../constants/filters';

const newFilterKey = 'budget_min';
const newFilterValue = 21000000;

function Filters({ filters }) {
    return Object.keys(filters).map((item) => <p key={item}>{item}</p>);
}

function MockedFilters({ filters, setFilter, clearFilters, closeModal, onSubmite }) {
    return (
        <>
            <Filters filters={filters} />
            <button data-testid="submit-reset-filters" onClick={onSubmite}>
                Submit reset filters
            </button>
            <button data-testid="close-modal" onClick={closeModal}>
                Set new Filters
            </button>
            <button data-testid="set-filters" onClick={() => setFilter(newFilterKey, newFilterValue)}>
                Set new Filters
            </button>
            <button data-testid="clear-filters" onClick={clearFilters}>
                Clear filters
            </button>
        </>
    );
}

jest.mock('../Filters/Filters.js', () => {
    return MockedFilters;
});
jest.mock('axios');

describe('Movies', () => {
    beforeEach(() => {
        window.scrollTo = jest.fn();
        axios.get.mockReturnValue(Promise.resolve({ data: mockMovies }));
    });
    it('should render all movies after success get request', async () => {
        let component;
        await act(async () => {
            component = await mount(
                <BrowserRouter>
                    <Movies />
                </BrowserRouter>
            );
        });
        component.update();
        expect(component.find('MovieCard').length).toBe(mockMovies.length);
    });

    it(`should set filters to passed key=${newFilterKey} value=${newFilterValue}`, async () => {
        let component;
        await act(async () => {
            component = await mount(
                <BrowserRouter>
                    <Movies />
                </BrowserRouter>
            );
        });
        component.find("[data-testid='set-filters']").simulate('click');
        expect(component.find('Filters').props().filters[newFilterKey]).toBe(newFilterValue);
    });

    it('should reset filters to initial state', async () => {
        let component;
        await act(async () => {
            component = await mount(
                <BrowserRouter>
                    <Movies />
                </BrowserRouter>
            );
        });
        component.find("[data-testid='set-filters']").simulate('click');
        expect(component.find('Filters').props().filters[newFilterKey]).toBe(newFilterValue);
        component.find("[data-testid='clear-filters']").simulate('click');
        expect(component.find('Filters').props().filters[newFilterKey]).toBe(FILTERS_INIT[newFilterKey]);
    });

    it('should get new filtered movies', async () => {
        let component;
        await act(async () => {
            component = await mount(
                <BrowserRouter>
                    <Movies />
                </BrowserRouter>
            );
        });
        let filteredMovies = mockMovies.filter((item) => item.budget <= newFilterValue);
        await act(async () => {
            axios.get.mockReturnValue(Promise.resolve({ data: filteredMovies }));
            await component.find("[data-testid='set-filters']").simulate('click');
            await component.find("[data-testid='submit-reset-filters']").simulate('click');
        });
        component.update();
        expect(component.find('MovieCard').length).toBe(filteredMovies.length);
    });

    it('should render NoData Component if movies are empty', async () => {
        let component;
        await act(async () => {
            component = await mount(
                <BrowserRouter>
                    <Movies />
                </BrowserRouter>
            );
        });
        let filteredMovies = mockMovies.filter((item) => item.budget > 999999999999999);
        await act(async () => {
            axios.get.mockReturnValue(Promise.resolve({ data: filteredMovies }));
            await component.find("[data-testid='set-filters']").simulate('click');
            await component.find("[data-testid='submit-reset-filters']").simulate('click');
        });
        component.update();
        expect(component.find('MovieCard').exists()).toBeFalsy();
        expect(component.find('NoData').exists()).toBeTruthy();
    });

    it('should toggle isFiltersHidden', async () => {
        let component;
        await act(async () => {
            component = await mount(
                <BrowserRouter>
                    <Movies />
                </BrowserRouter>
            );
        });
        component.update();
        expect(component.find("[data-testid='filter-modal']").at(0).props().hidden).toBeTruthy();
        component.find("[data-testid='filters-toggler']").simulate('click');
        component.update();
        expect(component.find("[data-testid='filter-modal']").at(0).props().hidden).toBeFalsy();
    });
});
