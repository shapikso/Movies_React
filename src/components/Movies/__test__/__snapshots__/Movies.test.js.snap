// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies should render correctly 1`] = `ReactWrapper {}`;
